Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js

```diff
@@ -0,0 +1,419 @@
+        cy.get("[data-testid=collection-date-1]").contains(
+            "2021-Dec-01, 4:07 PM"
+        );
+        cy.get("[data-testid=test-status]").should("be.visible");
+        cy.get("[data-testid=test-status]").contains("Corrected");
+        cy.get("[data-testid=result-date-1]").should("be.visible");
+        cy.get("[data-testid=result-date-1]").contains("2021-Dec-01, 4:07 PM");
+        cy.get("[data-testid=reporting-lab]").should("be.visible");
+        cy.get("[data-testid=reporting-lab]").contains("Fha");
+        cy.get("[data-testid=result-description]").should("be.visible");
+
+        cy.get("[data-testid=btnCheckAnotherTest]").should("be.visible");
+        cy.get("[data-testid=btnLogin]").should("be.visible");
+
+        cy.get("[data-testid=btnCheckAnotherTest]").click();
+        cy.get("[data-testid=phnInput]").should("be.visible");
+    });
+
+    it("Public Covid19 Test - Empty Results", () => {
+        const phn = "9735361219 ";
+        const dobYear = "1994";
+        const dobMonth = "June";
+        const dobDay = "9";
+        const dovYear = "2021";
+        const dovMonth = "January";
+        const dovDay = "20";
+
+        cy.intercept("GET", "**/v1/api/PublicLaboratory/CovidTests", (req) => {
+            req.reply({
+                fixture: "LaboratoryService/emptyCovidTest.json",
+            });
+        });
+
+        enterCovidTestPHN(phn);
+
+        select(dobYearSelector, dobYear);
+        select(dobMonthSelector, dobMonth);
+        select(dobDaySelector, dobDay);
+        select(dovYearSelector, dovYear);
+        select(dovMonthSelector, dovMonth);
+        select(dovDaySelector, dovDay);
+        clickCovidTestEnterButton();
+        cy.get("[data-testid=public-covid-test-result-form-title]").should(
+            "be.visible"
+        );
+        cy.get("[data-testid=btnCheckAnotherTest]").should("be.visible");
+        cy.get("[data-testid=btnLogin]").should("be.visible");
+    });
+
+    it("Public Covid19 Test - Data Mismatch", () => {
+        const phn = "9735361219 ";
+        const dobYear = "1994";
+        const dobMonth = "June";
+        const dobDay = "9";
+        const dovYear = "2021";
+        const dovMonth = "January";
+        const dovDay = "20";
+
+        cy.intercept("GET", "**/v1/api/PublicLaboratory/CovidTests", (req) => {
+            req.reply({
+                fixture: "LaboratoryService/dataMismatchCovidTest.json",
+            });
+        });
+
+        enterCovidTestPHN(phn);
+        select(dobYearSelector, dobYear);
+        select(dobMonthSelector, dobMonth);
+        select(dobDaySelector, dobDay);
+        select(dovYearSelector, dovYear);
+        select(dovMonthSelector, dovMonth);
+        select(dovDaySelector, dovDay);
+        clickCovidTestEnterButton();
+        cy.get("[data-testid=error-text-description]").should("be.visible");
+    });
+
+    it("Public Covid19 Test - Invalid Health Gateway Problem", () => {
```

Comment:
I would remove "Health Gateway Problem" from the name since it may confuse us later.